### PR TITLE
add repository dispatch to trigger test pipeline runs

### PR DESCRIPTION
**Pull request by hgarvison**
add repository dispatch to trigger test pipeline runs from confidential sidecars repo

Review in conjunction with: https://github.com/microsoft/confidential-sidecar-containers/pull/80

### CODE REVIEW

**Comment by DomAyre** [comment]
This change is okay on the basis that we're very careful with `ACC_REPO_DISPATCH_TOKEN` and that is has limited permissions. Is this token the same in this repo and in the sidecars repo? If we don't already, we should probably have two different tokens where the sidecars one can only comment on PR's and the examples one can only trigger a workflow (ideally only the sidecar workflows to be extra safe)

File: .github/workflows/test_attestation.yml [hunk 1]
```diff
@@ -58,15 +61,36 @@ jobs:
...
+    name: Create Pipeline Link Comment in PR
+    if: github.event_name == 'repository_dispatch'
+    runs-on: ubuntu-latest
+    env:
+      RUN_ID: ${{ github.run_id }}
+      PR_NUMBER: ${{ github.event.client_payload.pr_number }}
+    steps:
+      - name: Create Pipeline Link Repo Dispatch Request
+        run: |
+          curl -L \
```
Comment (ksayid) [inline .github/workflows/test_attestation.yml:86]: same comment from the other PR about adding the --fail flag to `curl`
https://github.com/microsoft/confidential-sidecar-containers/pull/80#discussion_r1472008543

File: .github/workflows/test_encrypted_filesystem.yml [hunk 2]
```diff
@@ -59,15 +62,36 @@ jobs:
...
+    name: Create Pipeline Link Comment in PR
+    if: github.event_name == 'repository_dispatch'
+    runs-on: ubuntu-latest
+    env:
+      RUN_ID: ${{ github.run_id }}
+      PR_NUMBER: ${{ github.event.client_payload.pr_number }}
+    steps:
+      - name: Create Pipeline Link Repo Dispatch Request
+        run: |
+          curl -L \
```
Comment (ksayid) [inline .github/workflows/test_encrypted_filesystem.yml:87]: same here

File: .github/workflows/test_key_release.yml [hunk 3]
```diff
@@ -59,15 +62,36 @@ jobs:
...
+    name: Create Pipeline Link Comment in PR
+    if: github.event_name == 'repository_dispatch'
+    runs-on: ubuntu-latest
+    env:
+      RUN_ID: ${{ github.run_id }}
+      PR_NUMBER: ${{ github.event.client_payload.pr_number }}
+    steps:
+      - name: Create Pipeline Link Repo Dispatch Request
+        run: |
+          curl -L \
```
Comment (ksayid) [inline .github/workflows/test_key_release.yml:87]: and here

**Review by ksayid — APPROVED** [review]
LGTM

**Pull request closed without merging**